src/test/java/com/faforever/api/security/FafTokenServiceTest.java

```diff
@@ -0,0 +1,120 @@
+package com.faforever.api.security;
+
+import com.faforever.api.config.FafApiProperties;
+import com.faforever.api.error.ApiExceptionWithCode;
+import com.faforever.api.error.ErrorCode;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.springframework.security.jwt.Jwt;
+import org.springframework.security.jwt.JwtHelper;
+import org.springframework.security.jwt.crypto.sign.MacSigner;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class FafTokenServiceTest {
+  private static final String TEST_SECRET = "banana";
+  private final MacSigner macSigner;
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+  private ObjectMapper objectMapper;
+  private FafTokenService instance;
+
+  public FafTokenServiceTest() {
+    this.macSigner = new MacSigner(TEST_SECRET);
+  }
+
+  @Before
+  public void setUp() {
+    objectMapper = new ObjectMapper();
+    objectMapper.registerModule(new JavaTimeModule());
+
+    FafApiProperties properties = new FafApiProperties();
+    properties.getJwt().setSecret(TEST_SECRET);
+
+    instance = new FafTokenService(objectMapper, properties);
+  }
+
+  @Test
+  public void createToken() throws Exception {
+    String token = instance.createToken(FafTokenType.REGISTRATION, Duration.ofSeconds(100), new HashMap<>());
```

Comment:
Use `Collections.emptyMap()` instead of `new HashMap<>()`, everywhere in this file.